engine/src/main.cpp

```diff
@@ -672,16 +672,22 @@ void bootstrap_main_loop()
         vector< SavedUnits >saved;
         vector< string > packedInfo;
 
+        bool loadedSavedGame = game_options.load_last_savegame;
         if (game_options.load_last_savegame) {
-            _Universe->AccessCockpit( k )->savegame->ParseSaveGame( savegamefile,
+            // `fatal` log level is used here so that it actually gets printed out without any buffering
+            // otherwise it'd be debug or info level
+            BOOST_LOG_TRIVIAL(fatal) << boost::format(" Loading %1% from Main") % savegamefile;
+            loadedSavedGame = _Universe->AccessCockpit( k )->savegame->ParseSaveGame( savegamefile,
                                                                     mysystem,
                                                                     mysystem,
                                                                     pos,
                                                                     setplayerXloc,
                                                                     credits,
                                                                     packedInfo,
                                                                     k );
-        } else {
+        }
+
+        if (!loadedSavedGame) {
```

Comment:
I think the original else should be left alone, and the `if (!loadedSavedGame) {` brought into the upper `if`, where it should at the very least log the fail and bail out immediately; it might make easier to revert to the savegame loading screen from here rather than from inside the `main_loop` called further below.

Reply:
I think this is for when you tell it to load a game via the command-line. It's not in the current call path that I'm working on where you load a game from the load screen.

For simplicity I broke this out so that it could revert to the previous behavior easier; so in this case if it fails then it goes on to load as if no saved game was requested to load.

Reply:
When the user tells VS to load a game via the command line, and it isn't able to load the game successfully, shouldn't VS bomb out and exit back to the command line with an error code? Whereas if the user tries to load a game via the GUI, failure to load should land the user right back at the same GUI screen, probably after an error dialog of some sort.